### PR TITLE
Fixes #762. Fix fromToBy & injectInto issues with same start and end with negative step. Add unit tests to Interval as well, to cover the scenario.

### LINKED ISSUE #762: IntInterval.fromToBy(n, n, -x) doesn't work as expected
Using eclipse collections 10.0.0

`IntInterval.fromToBy(2, 2, -1).toList()` raises an exception whereas we expect `[2]`.

 Here is the exception raised:
```
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at org.eclipse.collections.impl.list.primitive.IntInterval.lambda$toArray$7243f8d6$1(IntInterval.java:610)
	at org.eclipse.collections.impl.list.primitive.IntInterval.forEachWithIndex(IntInterval.java:324)
	at org.eclipse.collections.impl.list.primitive.IntInterval.toArray(IntInterval.java:610)
	at org.eclipse.collections.impl.list.mutable.primitive.IntArrayList.newList(IntArrayList.java:98)
	at org.eclipse.collections.impl.list.primitive.IntInterval.toList(IntInterval.java:900)
       [...]
```

`IntInterval.fromToBy(2,2,-2).forEach(i -> { System.out.println(i);}); ` prints an infinite sequence of decreasing integers whereas we expect just `2`.

This behaviour is not the same as `Interval.fromToBy(2,2,-2)` wich seems to be ok.

### PR DESCRIPTION
**Pull request by cguntur-bnym**
Signed-off-by: Chandra Guntur <chandra.guntur@bnymellon.com>

### CODE REVIEW

**Comment by motlin** [comment]
I never realized that this works:
`Interval.fromToBy(2, 2, 2)`

And this ought to work:
`Interval.fromToBy(2, 2, -2)`

But this is deliberately banned
`Interval.fromToBy(2, 2, 0)`

**Comment by cguntur-bnym** [comment]
This PR includes a change to injectInto() method as well, where I found the same issue. Added tests for that as well.

**Comment by nikhilnanivadekar** [comment]
Please break it up as separate commits or change the comment of the commit to mention that. It helps while preparing release notes to look at the commits and their comments. Thank you!

**Comment by cguntur-bnym** [comment]
Will run a commit amend and re-push over the weekend.

**Comment by nikhilnanivadekar** [comment]
@cguntur-bnym any updates on this? Also the commit message needs to state that it fixes which issue. Fixes #762 so that it will auto-close.

**Comment by cguntur-bnym** [comment]
@nikhilnanivadekar - please let me know if you need me to do anything else?

**Comment by nikhilnanivadekar** [comment]
@cguntur-bnym can you please rebase on to upstream/master . From the network graph below, your branch is a few commits behind the upstream
image

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/list/primitive/IntInterval.java [hunk 1]
```diff
@@ -339,13 +339,18 @@ public void forEach(IntProcedure procedure)
         this.each(procedure);
     }
 
+    public boolean goForward()
```
Comment (nikhilnanivadekar) [inline eclipse-collections/src/main/java/org/eclipse/collections/impl/list/primitive/IntInterval.java]: should be private

**Pull request merged**